### ISSUE TITLE
Can't replicate P2P to/from an iOS device on the same wifi network

### ISSUE THREAD

**Issue by joshblour** [couchbase/couchbase-lite-android#689]
I'm (happily :smile:) using CBL in an iOS app for P2P replication. I've started developing an android version but I'm not able to get replication working between android and iOS. Specifically: I'm unable to replicate (push+pull) from a Moto E (5.0.2) to an iPhone 6 (iOS 9.1) on on the same wifi network.

(Replication android-to-server (couchDB) works perfectly and I'm getting another android device this week to test android-to-android replication)

On the android, using Bonjour for discovery, I'm able to resolve the iPhones IP address and *start* the replication:
```
11-04 17:55:09.916 19835-19882/com.joshblour.proximator.android W/proximator-MyReplication: Pull Replication: http://p2p_v1:******@192.168.1.52:52021/proximator6 - active
11-04 17:55:09.916 19835-21347/com.joshblour.proximator.android D/Sync: [PullerInternal.waitForPendingFutures()] STARTED - thread id: 1813
11-04 17:55:10.416 19835-21347/com.joshblour.proximator.android D/Sync: batcher.waitForPendingFutures()
11-04 17:55:10.416 19835-21347/com.joshblour.proximator.android D/Sync: waitPendingFuturesCompleted()
11-04 17:55:10.416 19835-21347/com.joshblour.proximator.android D/Sync: calling future.get() on com.couchbase.lite.support.RemoteRequestRetry@8bb01c1
```
then nothing happens for 5 minutes (no callbacks from the replication's changelistener) until I receive this error:
```
11-04 18:00:10.028 19835-21343/com.joshblour.proximator.android E/RemoteRequest: io exception.  url: http://p2p_v1:******@192.168.1.52:52021/proximator6/_local/7a4e4d0e756e7811809a5294ee188b8e0d2abf36
11-04 18:00:10.028 19835-21343/com.joshblour.proximator.android E/RemoteRequest: java.net.SocketTimeoutException
```
Here's a gist with the full logs as well as the iphone logs for the same timeframe: https://gist.github.com/joshblour/46857abf0605941ea40f

**Comment by joshblour** [comment]
Ok, I've got it working now with SSL.

On the iOS side, I had `setAnonymousSSLIdentityWithLabel` setup on the listener.

I inserted the following code and change the destination URL schema to **https**:
```
        PersistentCookieStore cookieStore = database.getPersistentCookieStore();
        CouchbaseLiteHttpClientFactory cblHttpClientFactory = new CouchbaseLiteHttpClientFactory(cookieStore);
        cblHttpClientFactory.allowSelfSignedSSLCertificates();

        // set CouchbaseHttpClientFactory to Manager
        manager.setDefaultHttpClientFactory(cblHttpClientFactory);
```

Note: I was previously using **http** I would have imagined that would work even though SSL is setup on the listener. I'll leave this open incase you guys want to comment on it.

Also, I read in the comments that `allowSelfSignedSSLCertificates()` is not to be used in production. Is there another way that will work with `setAnonymousSSLIdentity` on iOS? The answers I've seen involve adding the certificate to your keystore but in this case, I don't know if we have access to the cert generated by `setAnonymousSSLIdentity`
